### ISSUE TITLE
PDP Cleanup- Basic Details Card, Tax Card & Inventory Card Updates

### ISSUE THREAD

**Issue by rymorgan** [reactioncommerce/reaction#2032]
See annotated .png for details. 

![inventory](https://cloud.githubusercontent.com/assets/20409254/24225489/07995a62-0f1e-11e7-87e2-3f952496d743.png)
![inventory-annotated](https://cloud.githubusercontent.com/assets/20409254/24225640/d5d92a2e-0f1e-11e7-8d4e-a4b7d5ac7fd4.png)

**Comment by rymorgan** [comment]
Related to https://github.com/reactioncommerce/reaction/issues/1647

**Comment by kieha** [comment]
@rymorgan question on this ticket: 
1. Is the `Allow Backorder` toggle what is currently the `Deny when out of stock` checkbox?
2. The `Quantity` field should be moved from the top of the details into the `Inventory Tracking` card?

<img width="394" alt="screen shot 2017-03-30 at 6 19 04 pm" src="https://cloud.githubusercontent.com/assets/19430095/24512105/53948c9e-1576-11e7-8827-28a4b35f53e3.png">

**Comment by rymorgan** [comment]
1. Yes, it's the allow back order toggle. Saying allow back order is a clearer more concise way of saying it. 
2. Yes, move quantity field. 
3. Absolutely, we should convert to react. @mikemurray want everything converted to react as we work on it. :)
4. Yes, let's do the card above taxable too. Here's a mock for that card. 

![pdp-details card](https://cloud.githubusercontent.com/assets/20409254/24520974/36e42aa0-153f-11e7-9349-7053f0618312.png)

**Comment by kieha** [comment]
@rymorgan cool. Thanks!

**Comment by kieha** [comment]
@rymorgan is there any need to expand the `Taxable` card when the expand is determined by the toggle?
My thinking: When you toggle the `Taxable` card is when you should see the fields nested under it.

**Comment by kieha** [comment]
@rymorgan this is what I have so far:

<img width="400" alt="screen shot 2017-04-05 at 8 14 33 pm" src="https://cloud.githubusercontent.com/assets/19430095/24717877/97c4d656-1a3c-11e7-961e-35d25e69b433.png">

Any nitpicks/suggestions for the UI part?

**Comment by mikemurray** [comment]
can you make a [WIP] PR for this to better track changes?

The switches are are the old blaze / CSS version. You should use the `<Switch>` component. Also our card component has the ability to show a switch by default. The switch component also has a label built in.

Switch component: https://github.com/reactioncommerce/reaction/blob/development/imports/plugins/core/ui/client/components/switch/switch.js#L5

quick example: 

Using the settings card will have everything you need to one of those settings panel, Its a combination of our `Card` component and a `Switch` so its pretty much a drop-in-repacement for `Card`.

https://github.com/reactioncommerce/reaction/blob/development/imports/plugins/included/social/client/components/settings.js#L71
```
          <SettingsCard
            key={index}
            i18nKeyTitle={`settings.${provider.name}`}
            expandable={true}
            onExpand={this.props.onSettingExpand}
            expanded={this.props.preferences[provider.name]}
            title={provider.name}
            name={provider.name}
            enabled={this.props.socialSettings.settings.apps[provider.name].enabled}
            icon={provider.icon}
            onSwitchChange={this.props.onSettingEnableChange}
          >
            <Form
              schema={SocialPackageConfig}
              doc={doc}
              docPath={`settings.public.apps.${provider.name}`}
              hideFields={[
                `settings.public.apps.${provider.name}.enabled`
              ]}
              name={`settings.public.apps.${provider.name}`}
              onSubmit={this.handleSubmit}
            />
          </SettingsCard>
        );
```


https://github.com/reactioncommerce/reaction/blob/development/imports/plugins/core/dashboard/client/components/toolbar.js#L65
```
      <Switch
        i18nKeyLabel="app.editMode"
        i18nKeyOnLabel="app.editMode"
        label={"Edit Mode"}
        onLabel={"Edit Mode"}
        checked={!this.props.isPreview}
        onChange={this.onViewContextChange}
      />
```

**Comment by rymorgan** [comment]
Thanks, @mikemurray -- here are my visual notes. This might end up being fixed when we switch over to Mike's components. 

<img width="617" alt="2032-2" src="https://cloud.githubusercontent.com/assets/20409254/24719236/deb5a54e-19ed-11e7-8bc3-572cb6783e26.png">
<img width="929" alt="2032" src="https://cloud.githubusercontent.com/assets/20409254/24719235/deb21e2e-19ed-11e7-8544-76b53c0e7b15.png">

**Comment by rymorgan** [comment]
I also noticed that when you have all of the cards closed and go to open one of the variant options it opens up all the cards in the whole sidebar. It should only open that variant option only.

**Comment by kieha** [comment]
@rymorgan the `Variant Options` collapsible card request is in issue #2031. I'll add it when I'm done with this and take it up.

**Comment by kieha** [comment]
Status update:
Almost done with the ticket. Design part done. What's left is implementing some functionality.

**Comment by kieha** [comment]
Update:
- All functionality implemented.
Pending:
- Fixing reactivity issues
- Implement update quantity field with child variant quantities

**Comment by kieha** [comment]
Closed by #2086
